mods/cnc/maps/nod02a/nod02a.lua

```diff
@@ -0,0 +1,296 @@
+-- standard spawn point
+Spawnpoint = { waypoint27.Location }
+
+-- Actors that activate a trigger
+DfndActorTriggerActivator = { Refinery, Barracks, Powerplant, Yard}
+Atk3ActorTriggerActivator = { Guard1, Guard2, Guard3, Guard4, Guard5, Guard6, Guard7 }
+
+-- Cell Triggers
+Atk1CellTriggerActivator = {CPos.New(45,37), CPos.New(44,37), CPos.New(45,36), CPos.New(44,36), CPos.New(45,35), CPos.New(44,35), CPos.New(45,34), CPos.New(44,34)}
+Atk4CellTriggerActivator = {CPos.New(50,47), CPos.New(49,47), CPos.New(48,47), CPos.New(47,47), CPos.New(46,47), CPos.New(45,47), CPos.New(44,47), CPos.New(43,47), CPos.New(42,47), CPos.New(41,47), CPos.New(40,47), CPos.New(39,47), CPos.New(38,47), CPos.New(37,47), CPos.New(50,46), CPos.New(49,46), CPos.New(48,46), CPos.New(47,46), CPos.New(46,46), CPos.New(45,46), CPos.New(44,46), CPos.New(43,46), CPos.New(42,46), CPos.New(41,46), CPos.New(40,46), CPos.New(39,46), CPos.New(38,46)}
+
+
+-- ************************** Helper functions Start ************************
+-- returns a table with filtered actors who are alive and from the specified type and owner
+getActors = function(owner, units)	
+    local maxUnits = 0
+    local counter = 1
+	local actors = {}		
+    for type, count in pairs(units) do
+        local globalActors = Map.ActorsInBox(Map.TopLeft, Map.BottomRight,	function(actor)
+            return actor.Owner == owner and tostring(actor.Type) == type and actor.IsDead == false
+        end)
+        if #globalActors < count then
+            maxUnits = #globalActors
+        else
+            maxUnits = count
+        end        
+		for i = 1, maxUnits, 1 do	
+	        actors[counter] = globalActors[i]	
+            counter = counter + 1		    
+	    end
+    end
+	return actors
+end
+
+
+-- set ondamaged trigger for array of actors
+OnAnyDamaged = function(actors, func)
+    for key in pairs(actors) do
+        Trigger.OnDamaged(actors[key], func)
+    end
+end
+
+-- check if player has buildings
+CheckForBase = function(player, buildings)
+	local baseBuildings = Map.ActorsInBox(Map.TopLeft, Map.BottomRight, function(actor)
+        local found = false
+        for i = 1, #buildings, 1 do
+            if actor.Type == buildings[i] then
+                found = true
+            end
+        end
+		return actor.Owner == BadGuy and found
+	end)        
+	return #baseBuildings >= 3
+end
+-- ************************** Helper functions End ********************************************
+
+-- ************************** ATK7 start **********************************************
```

Comment:
These comments are unnecessary. Golden rule for comments: don't describe what you're doing (that should be self-explanatory from the code), but describe *why* you're doing it (if necessary, most of the time it's obvious as well)